ruby-challenges.rb

```diff
@@ -11,21 +18,116 @@
 # Expected output: ['tea', 'water', 'soda water']
 
 
+def filter_letter(arr, letter)
+    arr.filter {|word| word.include?(letter)}
+end
+
+filtered_arr = filter_letter(beverages_array,filter_letter_o)
+
+p filtered_arr
+
+
+
 # -------------------2) Create a method that takes in a hash and returns one array with all the hash values at their own index and in alphabetical order. No nested arrays. Use the test variable provided.
 # HINT: Google 'ruby get rid of nested arrays'
 
-us_states = { northwest: ['Washington', 'Oregon', 'Idaho'], southwest: ['California', 'Arizona', 'Nevada'], notheast: ['Maine', 'New Hampshire', 'Rhode Island'] }
+us_states = { northwest: ['Washington', 'Oregon', 'Idaho'], southwest: ['California', 'Arizona', 'Nevada'], northeast: ['Maine', 'New Hampshire', 'Rhode Island'] }
 # Expected output: ['Arizona', 'California', 'Idaho', 'Maine', 'Nevada', 'New Hampshire', 'Oregon', 'Rhode Island', 'Washington'] 
 
+# Pseudo Code:
+# Process: I created two methods one that can get rid of the nested arrays and one that can sort them alphabetically so that we can get the expected outcome. 
+# Flatten Hash: I created a method that takes the hash as an input and uses the .flatten method to extract the values from the nested arrays.
+# newArr: I define the new array to be used for refrence later
+# Organize: I create a method that uses the .map method to iterate through our new array and pass our states through and use the .sort method to sort through our states and alphabetize them correctly.
+# DIsplay: To correctly display our outcome, we refrence our organizer method and use our flattened array defined as newArr to produce expected outcome. 
+
+
+
+def flatten_hash(hash)
+   hash.values.flatten
+end
+
+newArr = flatten_hash(us_states)
+
+def organize(arr)
+    arr.map { |states| states}.sort
+end    
```

Comment:
How can we combine all of this into one method? By the end of line 47, you have 1 array with all the values in it. From there, you can just add .sort to the end to sort them in alphabetical order!